### ISSUE TITLE
PHP8.1: Deprecated warning caused by the plugin

### ISSUE THREAD

**Issue by idflood** [nosilver4u/ewww-image-optimizer#291]
When enabling this plugin on php8.1 website with Debug mode on there are "Deprecated" warnings shown on the backend.

**To Reproduce**
1. Install Wordpress on php8.1
2. Enable the EWWW Image Optimizer module
3. The deprecated warnings are shown at the top of the page.
4. If you go to /wp-admin/options-general.php?page=ewww-image-optimizer-options there are a few more deprecated warnings.

**Expected behavior**
There should be no warning. Disabling the module or downgrading to php8 remove these errors.

**Screenshots**

![Screenshot 2022-11-24 at 11 30 07](https://user-images.githubusercontent.com/197418/203761577-29d7d64e-05a4-4999-894b-ddcb394fcced.png)

For reference here are the errors shown in the plugin option page:

```
Deprecated: strstr(): Passing null to parameter #1 ($haystack) of type string is deprecated in /var/www/app/wp-includes/functions.php on line 1145

Deprecated: stripos(): Passing null to parameter #1 ($haystack) of type string is deprecated in /var/www/app/wp-includes/functions.php on line 1152

Deprecated: stripos(): Passing null to parameter #1 ($haystack) of type string is deprecated in /var/www/app/wp-includes/functions.php on line 1155

Deprecated: strpos(): Passing null to parameter #1 ($haystack) of type string is deprecated in /var/www/app/wp-includes/functions.php on line 1162

Deprecated: strpos(): Passing null to parameter #1 ($haystack) of type string is deprecated in /var/www/app/wp-includes/functions.php on line 1165

Deprecated: strstr(): Passing null to parameter #1 ($haystack) of type string is deprecated in /var/www/app/wp-includes/functions.php on line 1145

Deprecated: stripos(): Passing null to parameter #1 ($haystack) of type string is deprecated in /var/www/app/wp-includes/functions.php on line 1152

Deprecated: stripos(): Passing null to parameter #1 ($haystack) of type string is deprecated in /var/www/app/wp-includes/functions.php on line 1155

Deprecated: strpos(): Passing null to parameter #1 ($haystack) of type string is deprecated in /var/www/app/wp-includes/functions.php on line 1162

Deprecated: strpos(): Passing null to parameter #1 ($haystack) of type string is deprecated in /var/www/app/wp-includes/functions.php on line 1165

Deprecated: strstr(): Passing null to parameter #1 ($haystack) of type string is deprecated in /var/www/app/wp-includes/functions.php on line 1145

Deprecated: stripos(): Passing null to parameter #1 ($haystack) of type string is deprecated in /var/www/app/wp-includes/functions.php on line 1152

Deprecated: stripos(): Passing null to parameter #1 ($haystack) of type string is deprecated in /var/www/app/wp-includes/functions.php on line 1155

Deprecated: strpos(): Passing null to parameter #1 ($haystack) of type string is deprecated in /var/www/app/wp-includes/functions.php on line 1162

Deprecated: strpos(): Passing null to parameter #1 ($haystack) of type string is deprecated in /var/www/app/wp-includes/functions.php on line 1165
```

**Comment by nosilver4u** [comment]
Thanks for the report, technically that's a "bug" in WP core and should be fixed there.
However, we may be able to modify our usage of the WP core add_query_arg() function to accommodate that bug, so I'll look into it when I have time.

I use the term "bug" loosely, as you can see what core devs have said regarding deprecation notices here: https://make.wordpress.org/core/2022/01/10/wordpress-5-9-and-php-8-0-8-1/

In short, deprecation notices are neither errors nor warnings, and should not ever be displayed on a production website. If they ARE shown on a production website running EWWW IO, then the PHP error handling should be modified to suppress notices and informational messages.

**Comment by powernic** [comment]
@nosilver4u This error occurs in this plugin because you are using incorrect WP core functions. But I fixed It #294. Please check it.

**Comment by nosilver4u** [comment]
Fixed all occurrences of add_query_arg() to use an empty string instead of null.
@powernic Also changed our usage of add_submenu_page() to pass an empty string instead of null so that PHP gets the type of variable it is expecting.

**Comment by powernic** [comment]
@nosilver4u Thx!
